var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import llnl.util.tty as tty
+import os
+
+
+class Upcxx(Package):
+    """UPC++ is a C++ library that supports Partitioned Global Address Space
+    (PGAS) programming, and is designed to interoperate smoothly and
+    efficiently with MPI, OpenMP, CUDA and AMTs. It leverages GASNet-EX to
+    deliver low-overhead, fine-grained communication, including Remote Memory
+    Access (RMA) and Remote Procedure Call (RPC)."""
+
+    homepage = "https://upcxx.lbl.gov"
+
+    version('2019.3.2', '844722cb0e8c0bc649017fce86469457')
+
+    variant('cuda', default=False, 
+            description='Builds a CUDA-enabled version of UPC++')
+    depends_on('cuda', when='+cuda')
+
+
+    def detect_scheduler():
```

Comment:
Move this outside the class.  Putting a method with no parameters here exposes it as an instance method on `Upcxx`, which isn't exactly wrong, as it won't be called that way, but Python will just complain about the lack of a `self` parameter if it's ever invoked.  So, it reads oddly.

Example:

```python
#!/usr/bin/env python
#
# What happens when you have an instance method with no `self` parameter?
#

class Test(object):
    def method():
        print("method was called!")
        return 1

    x = method()

# this should be 1
print(Test.x)

# this will fail
test = Test()
test.method()
```

```console
(spackbook):python$ ./no-self-instance-method.py
method was called!
1
Traceback (most recent call last):
  File "./no-self-instance-method.py", line 23, in <module>
    test.method()
TypeError: method() takes no arguments (1 given)
```